### PR TITLE
[17.0][FIX] repair_type: Remove duplicated destination location for added parts

### PR DESCRIPTION
**Pull request by ppyczko**
The default 'Locations' for a Repair operation in Odoo core are as follows:

imagen

After some tests, I observed that the field '**Default Destination Location**' is misleading, as it doesn't correspond to the destination location of the repaired product. Instead, **it corresponds to the destination location of the added parts.**

This can be seen in the following clip:

https://github.com/user-attachments/assets/41282925-818a-4e55-9745-ac08bfc97db9

As we can see, the 'Pedal Bin' product is used as an added part to the repair order. The stock move that is being created transfers the product to 'Virtual Locations/Production', which corresponds to the 'Default Destination Location' field on the related stock picking type. On the other hand, the 'Customizable Desk' product is being sent back to 'WH/Stock', which corresponds to the 'Default Source Location' field.

In contrast, the 'Locations' for a Repair operation that are defined in this repair_type module are the following:

imagen

Basically, the new field 'Default Add Destination Location' overrides the functionality of the 'Default Destination Location' field. 

Therefore, the changes in this PR intend to remove this field. Additionally, the tests have been refactored to be more reusable.

### CODE REVIEW

**Comment by ppyczko** [comment]
Fix for module repair_type 17.0

cc https://github.com/APSL 163691

@miquelalzanillas @lbarry-apsl @mpascuall @peluko00 @javierobcn @BernatObrador please review

**Comment by ppyczko** [comment]
Hi @LoisRForgeFlow, when you have the time, could you please take a look at this? I’d love to hear your input. Thank you in advance!

File: repair_type/models/repair.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Copyright 2024 Patryk Pyczko (APSL-Nagarro)<ppyczko@apsl.net>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class RepairOrder(models.Model):
+    _inherit = "repair.order"
+
+    product_location_dest_id = fields.Many2one(
```
Comment (LoisRForgeFlow) [inline repair_type/models/repair.py]: I agree overall with the change :+1: 

However, being OCA we need to be as modular as possible. Since this fields doesn't do anything without a future `repair_picking_after_done` extension, I think we would need to move that to a glue module: something like `repair_type_picking_after_done` where the field is added and affects the picking after done behavior.

Does this sound reasonable?

Comment (ppyczko) [reply repair_type/models/repair.py]: Thank you for your feedback @LoisRForgeFlow, your suggestion makes a lot of sense. 

I’ve applied the suggested changes by removing the fields related to the repaired product's location out of repair_type and will move them into a separate module as suggested. Let me know if there’s anything else to adjust!

**Review by LoisRForgeFlow — APPROVED** [review]
Great, this one looks ready then :+1:

**Review by peluko00 — APPROVED** [review]
LGTM

**Comment by ppyczko** [comment]
> Great, this one looks ready then 👍

Thank you again, @LoisRForgeFlow! Can you merge it, or should I ping someone?

**Comment by LoisRForgeFlow** [comment]
All good, let's merge!

/ocabot merge minor
